### PR TITLE
Kv version

### PR DESCRIPTION
**Pull request by mrisher**
Dan and Alex built the KV variant

### CODE REVIEW

File: mta-sts.md [hunk 1]
```diff
@@ -561,16 +610,15 @@ MTA-STS policy indicator TXT RR:
 _mta-sts.example.com.  IN TXT "v=STSv1; id=20160831085700Z;"
 ~~~~~~~~~
 
-MTA-STS Policy JSON served as the response body at
-https://mta-sts.example.com/.well-known/mta-sts.json:
+MTA-STS Policy file served as the response body at
+https://mta-sts.example.com/.well-known/mta-sts.policy:
```
Comment (prbinu) [inline mta-sts.md]: Should we mention HTTP response content-type header:
Content-Type: text/plain

**Comment by danmarg** [comment]
I have an open issue on this. I will take a stab.

File: mta-sts.md [hunk 2]
```diff
@@ -148,24 +148,22 @@ An example TXT record is as below:
 
 `_mta-sts.example.com.  IN TXT "v=STSv1; id=20160831085700Z;"`
 
-The formal definition of the `_mta-sts` TXT record, defined using [@!RFC5234],
+The formal definition of the `_mta-sts` TXT record, defined using [@!RFC7405],
 is as follows:
 
-    sts-text-record = sts-version *WSP field-delim *WSP sts-id
+    sts-text-record = sts-version field-delim sts-id
                       [field-delim [sts-extensions]]
```
Comment (prbinu) [inline mta-sts.md]: We can simplify this my replacing [field-delim [sts-extensions]] with 
*(field-delim sts-extension)

and remove line 166:
sts-extensions  = sts-extension *(field-delim sts-extension)
                        [field-delim]                      ; extension fields

File: mta-sts.md [hunk 3]
```diff
@@ -198,32 +196,80 @@ This JSON object contains the following key/value pairs:
...
+    sts-policy-mx-field      = %s"mx"
+
+    sts-policy-mx-value      = 1*(ALPHA / DIGIT / "_" / "-" / "~" / ".")
+
+    sts-policy-max-age       = sts-policy-max-age-field field-delim
+                               sts-policy-max-age-value
+
+    sts-policy-max-age-field = %s"max_age"
+
+    sts-policy-max-age-value = 1*10DIGIT
```
Comment (prbinu) [inline mta-sts.md]: 1*10DIGIT ==> 1*10(DIGIT)

File: mta-sts.md [hunk 4]
```diff
@@ -198,32 +196,80 @@ This JSON object contains the following key/value pairs:
...
 
+The formal definition of the policy resource, defined using [@!RFC7405], is as
+follows:
+
+    sts-policy-record        = sts-policy-version line-delim
+                               sts-policy-mode line-delim
+                               sts-policy-mx line-delim [sts-policy-mx line-delim]
+                               sts-policy-max-age
+
+    line-delim               = %x0d %x0A                          ; "CRLF"
```
Comment (prbinu) [inline mta-sts.md]: replace line-delim with CRLF?

File: mta-sts.md [hunk 5]
```diff
@@ -198,32 +196,80 @@ This JSON object contains the following key/value pairs:
...
+mx: backupmx.example.com
+max_age: 123456
 ```
 
+The formal definition of the policy resource, defined using [@!RFC7405], is as
+follows:
+
+    sts-policy-record        = sts-policy-version line-delim
+                               sts-policy-mode line-delim
+                               sts-policy-mx line-delim [sts-policy-mx line-delim]
```
Comment (prbinu) [inline mta-sts.md]: replace with sts-policy-mx CRLF *(sts-policy-mx CRLF) ?

File: mta-sts.md [hunk 6]
```diff
@@ -198,32 +196,80 @@ This JSON object contains the following key/value pairs:
...
+                               sts-policy-max-age
+
+    line-delim               = %x0d %x0A                          ; "CRLF"
+
+    field-delim              = ":" WSP*
```
Comment (prbinu) [inline mta-sts.md]: *WSP

File: mta-sts.md [hunk 7]
```diff
@@ -198,32 +196,80 @@ This JSON object contains the following key/value pairs:
...
+    sts-policy-mode-field    = %s"mode"
+
+    sts-policy-model-value   =  %s"report" / %s"enforce"
+
+    sts-policy-mx            = sts-policy-mx-field field-delim
+                               sts-policy-mx-value
+
+    sts-policy-mx-field      = %s"mx"
+
+    sts-policy-mx-value      = 1*(ALPHA / DIGIT / "_" / "-" / "~" / ".")
```
Comment (prbinu) [inline mta-sts.md]: Do we have a case that requires '~' ?

File: mta-sts.md [hunk 8]
```diff
@@ -198,32 +198,81 @@ This JSON object contains the following key/value pairs:
...
-  "mode": "enforce",
-  "mx": [".mail.example.com"],
-  "max_age": 123456
-}
+version: STSv1
+mode: enforce
+mx: mail.example.com
+mx: .example.net
+mx: backupmx.example.com
+max_age: 123456
```
Comment (danmarg) [inline mta-sts.md]: This was discussed on the UTA list. I couldn't really tell what the consensus was; Viktor suggested removing the ":", whereas Alexey pointed out that with the : you could use a generic RFC2822 message header parser. 

I think it's not a big deal either way. Alexey's argument convinces me, mostly.

File: mta-sts.md [hunk 9]
```diff
@@ -561,16 +610,15 @@ MTA-STS policy indicator TXT RR:
...
-  "mx": ["mx1.example.com", "mx2.example.com"],
-  "max_age": 12345678
-}
-
+version: STSv1
+mode: report
+mx: mx1.example.com
+mx: mx2.example.com
+mx: mx.backup-example.com
+max_age: 12345678
```
Comment (danmarg) [inline mta-sts.md]: Replied above.

File: mta-sts.md (hunk 3 above)
Comment (danmarg) [inline mta-sts.md]: I think this is valid as-is, no? https://tools.ietf.org/html/rfc5234#section-3.7

File: mta-sts.md (hunk 4 above)
Comment (danmarg) [inline mta-sts.md]: Good call. Done.

File: mta-sts.md (hunk 5 above)
Comment (danmarg) [inline mta-sts.md]: https://tools.ietf.org/html/rfc5234#section-3.8

[sts-policy-mx CRLF] 

and 

*1(sts-policy-mx CRLF)

are the same, no?

File: mta-sts.md (hunk 6 above)
Comment (danmarg) [inline mta-sts.md]: Done.

File: mta-sts.md (hunk 7 above)
Comment (danmarg) [inline mta-sts.md]: Heh, good catch. I got that from the valid URI characters (https://tools.ietf.org/html/rfc3986), but ~ is not valid in a hostname, only in the path. Fixed.

**Comment by danmarg** [comment]
d242d89

File: mta-sts.md (hunk 8 above)
Comment (prbinu) [inline mta-sts.md]: 👍

File: mta-sts.md (hunk 3 above)
Comment (prbinu) [inline mta-sts.md]: yeah, i thought the latter would be more readable.

File: mta-sts.md (hunk 5 above)
Comment (prbinu) [inline mta-sts.md]: What you mentioned is correct. But my comment was slightly different:
*(sts-policy-mx CRLF)

File: mta-sts.md (hunk 2 above)
Comment (danmarg) [inline mta-sts.md]: Done.

File: mta-sts.md (hunk 5 above)
Comment (danmarg) [inline mta-sts.md]: Oh, but is a policy with zero MX records valid? I think it should not be...

File: mta-sts.md (hunk 3 above)
Comment (danmarg) [inline mta-sts.md]: Done.

File: mta-sts.md (hunk 1 above)
Comment (danmarg) [inline mta-sts.md]: Done.

File: mta-sts.md (hunk 5 above)
Comment (prbinu) [inline mta-sts.md]: Let me be very specific:
Replace this:
 sts-policy-mx line-delim [sts-policy-mx line-delim]
with
sts-policy-mx CRLF *(sts-policy-mx CRLF)

File: mta-sts.md (hunk 1 above)
Comment (davidillsley) [inline mta-sts.md]: if it's going to be text/plain, can it be a .txt file so default config works most places (eg nginx and S3 serves a file called mta-sts.policy as application/octet-stream by default)?

File: mta-sts.md (hunk 5 above)
Comment (danmarg) [inline mta-sts.md]: Oh, well, why not just do 

1*(sts-policy-mx CRLF) instead?

File: mta-sts.md (hunk 1 above)
Comment (danmarg) [inline mta-sts.md]: Dunno. I don't see why not, and it's a small change vs the k/v change. Done.

File: mta-sts.md (hunk 9 above)
Comment (prbinu) [inline mta-sts.md]: 👍

File: mta-sts.md [hunk 10]
```diff
@@ -198,32 +194,78 @@ This JSON object contains the following key/value pairs:
...
 
+The formal definition of the policy resource, defined using [@!RFC7405], is as
+follows:
+
+    sts-policy-record        = sts-policy-version CRLF
+                               sts-policy-mode CRLF
+                               1*(sts-policy-mx CRLF)
+                               sts-policy-max-age
+
+    field-delim              = ":" *WSP
```
Comment (prbinu) [inline mta-sts.md]: Would it be better to have at least one space between ":" and value?

Replace:
field-delim              = ":" *WSP
with 
field-delim              = ":" 1*WSP

File: mta-sts.md [hunk 11]
```diff
@@ -148,24 +148,19 @@ An example TXT record is as below:
...
+The formal definition of the `_mta-sts` TXT record, defined using [@!RFC7405],
 is as follows:
 
-    sts-text-record = sts-version *WSP field-delim *WSP sts-id
-                      [field-delim [sts-extensions]]
+    sts-text-record = sts-version field-delim sts-id
+                      *(field-delim sts-extension) [field-delim]
 
-    field-delim     = %x3B                               ; ";"
+    field-delim     = *WSP ";" *WSP
```
Comment (prbinu) [inline mta-sts.md]: Replace:
field-delim     = *WSP ";" *WSP
with
field-delim     = *WSP ";" 1*WSP

**Pull request merged**